### PR TITLE
SyncBot: allow a whisper's text to archive a thread

### PR DESCRIPTION
**Pull request by sqweelygig**
* Made SyncBot's createComment into a more multi-functional processCommand
* Made Front translator methods for archiveThread
* Used the above to process #archive
* Added stack trace to the debug output while I was there
* Pulled various almost identical functions into the new processCommand

Change-Type: patch
Connects-To: #353 

---
##### Contributor checklist

- [x] Build output been rebuilt and tested
- [ ] Introduces security considerations
- [x] Tests are included
- [x] Documentation is added or changed
- [ ] Affects the development, build or deployment processes of the component

##### Reviewer Guidelines
- When submitting a review, please pick:
  - '*Approve*' if this change would be acceptable in the codebase (even if there are minor or cosmetic tweaks that could be improved).
  - '*Request Changes*' if this change would not be acceptable in our codebase (e.g. bugs, changes that will make development harder in future, security/performance issues, etc).
  - '*Comment*' if you don't feel you have enough information to decide either way (e.g. if you have major questions, or you don't understand the context of the change sufficiently to fully review yourself, but want to make a comment)
---
---- Autogenerated Waffleboard Connection: Connects to #353

### CODE REVIEW

File: lib/bots/syncbot.ts [hunk 1]
```diff
@@ -186,20 +190,53 @@ export class SyncBot extends ProcBot {
...
-							thread: threadId,
-						}, messenger, data)
-						// Ignore the actual response from the messenger, SyncBot only cares that it's happened.
-						.return();
+						const flow = { service: to.service, flow: to.flow, thread: threadId };
+						return SyncBot.processCommand(flow, messenger, data, MessengerAction.UpdateTags)
+						.return(response);
+					}
+					// The correct action was to do nothing, so pass the details along to the next thing
+					return Promise.resolve(response);
```
Comment (hedss) [inline lib/bots/syncbot.ts]: Only need to `return response;`, not `Promise.resolve`.

File: lib/bots/syncbot.ts [hunk 2]
```diff
@@ -186,20 +190,53 @@ export class SyncBot extends ProcBot {
...
+							} else if (error instanceof TranslatorError && error.code === TranslatorErrorCode.EmitUnsupported) {
+								// If the service does not support the action then do not panic
+								return threadDetails;
+							} else {
+								return response;
+							}
+						});
 					}
+					// The correct action was to do nothing, so pass the details along to the next thing
+					return Promise.resolve(threadDetails);
```
Comment (hedss) [inline lib/bots/syncbot.ts]: Only need to `return threadDetails;`.

File: lib/bots/syncbot.ts [hunk 3]
```diff
@@ -186,20 +190,53 @@ export class SyncBot extends ProcBot {
...
+				})
+				// We've got no more actions to perform, but ProcBot only expects promise resolution, no actual payload
+				// Explicitly bookending the promise chain enables each .then() above to be treated atomically
+				.then((_threadDetails: MessengerEmitResponse) => {
+					return Promise.resolve();
```
Comment (hedss) [inline lib/bots/syncbot.ts]: `return;`

**Comment by sqweelygig** [comment]
#close #archive

**Pull request merged**